pkg/lib/cockpit-components-checkbox-select.tsx

```diff
@@ -126,8 +123,10 @@ const CheckboxSelectBase: React.FunctionComponent<CheckboxSelectProps> = ({
   };
 
   const _onSelect = (event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
```

Comment:
Naïve question wrt. that PF bug: Why can't we just declare this as `value: T` here and do away with the `as`? That seems kind of a lie, as you explicitly (originally) introduced the values as `unknown` as at least some code would throw objects at it (like metrics.jsx). `string as { value: date, content: text }` isn't going to end up well, no?

Reply:
(It's systemd/logs.jsx that throws objects into a Select, not metrics. See https://github.com/cockpit-project/cockpit/blob/main/pkg/systemd/logs.jsx#L57.)

Reply:
> Why can't we just declare this as value: T here and do away with the as?

If we do that, we get a type error further down with `onSelect={_onSelect}`.

We can put the "as" workaround on that property assignment, I would agree that it is the better place. But it's so much more noisy:

```
onSelect={_onSelect as (event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => void}
```

Reply:
Ah right, the object-y one in metrics.jsx is a `Select`, not a `CheckboxSelect`.

Reply:
> Ah right, the object-y one in metrics.jsx is a `Select`, not a `CheckboxSelect`.

Hmm, no, the SimpleSelect uses numbers.

Reply:
> We can put the "as" workaround on that property assignment, I would agree that it is the better place. But it's so much more noisy:

Agreed on IRC to just put a "@ts-expect-error" on this. Nice!